### DISCUSSION TITLE
Issues with SMTP and Office 365 (Exhange Online)

### DISCUSSION THREAD

**Discussion by SamuelQuinones** [pocketbase/pocketbase#1217] [Q&A]
I am trying to setup pocketbase to use smtp to send emails,

I have a custom domain that uses microsoft's exchange online to route / handle email. I know the domain is properly configured to use SMTP because other applications use it just fine.

But with pocketbase, each time i try to send the test email i get the error: 504 5.7.4 Unrecognized authentication type.

I've tried with multiple domains on the same exchange tenant, and a domain on a completely separate tenant - same results.

I've also tried in pocketbase 7 and pocketbase 9 just to see if it was maybe a newer version issue - same results

The domains in question do have dmarc and dkim enabled if that is in any way relevant

**Comment by ganigeorgiev** [comment]
@SamuelQuinones Are you using the "Auto (StartTLS)" or "Always" TLS encryption setting?

Maybe is conflicting with the default configured TLS version (by default I think is TLS 1.2), I'm no really sure. I'm not familiar with the microsoft ecosystem and I'm not sure how to test it. By google searching the error I stumbled on https://learn.microsoft.com/en-us/answers/questions/788460/504-574-unrecognized-authentication-type-with-o365.html, which seems related to https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/opt-in-exchange-online-endpoint-for-legacy-tls-using-smtp-auth and https://office365itpros.com/2021/08/19/exchange-online-to-introduce-legacy-smtp-endpoint-in-2022/.

**Reply by SamuelQuinones** [reply]
I’ve tried both;
- auto is what gives the 504 error
- Always gives an error right away about a handshake failing

I stumbled across that first article as well, I tried enabling legacy auth but no change, I can try again - maybe it didn’t propagate

**Reply by ganigeorgiev** [reply]
Could you confirm that `SMTP AUTH` is enabled per https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/authenticated-client-smtp-submission?

**Reply by SamuelQuinones** [reply]
can confirm SMTP auth is enabled

**Reply by ganigeorgiev** [reply]
@SamuelQuinones I'm not sure what could be the issue.

I suspect that it could be related to the fact that we are using `PLAIN` auth (aka. [rfc 4616](https://www.rfc-editor.org/rfc/rfc4616)), but while searching for the issue I found that microsoft probably works with `LOGIN` auth (which is not supported out of the box by the std `net/smtp` package) but cannot find information in the microsoft docs where this is clearly stated.

Just for the tests, if possible, could you try setupping a mailer with [nodemailer](https://nodemailer.com/about/#example) for example and see if you can reproduce the issue? To confirm that the auth method is the culprit, please try with both `authMethod: 'PLAIN'` and `authMethod: 'LOGIN'` to see which will succeed.

**Reply by SamuelQuinones** [reply]
Sorry for the delayed update -

I did a test with nodemailer, with `authMethod: 'PLAIN'` i get that same unrecognized authentication type error, but with `authMethod: "LOGIN"` the email sends without issue

**Reply by ganigeorgiev** [reply]
@SamuelQuinones Thank you for testing. In this case, we'll have to implement the `LOGIN` auth method (and maybe add a new "Auth method" option in the Admin UI SMTP settings). I'll try to apply the required changes sometime this week so that it could be available with the next release.

**Reply by SamuelQuinones** [reply]
Glad I could help! if you need me to test anything that would fix this - since i have the MS tenants - I'd be more than happy to help with that too!

**Reply by ganigeorgiev** [reply]
@SamuelQuinones Just to let you that I've added in `master` (https://github.com/pocketbase/pocketbase/commit/1f45b858a671ed2f12ee7014a57d16c9fc1c2dd2) support for the `LOGIN` auth method and it will be available with the next release.
I don't have a office365 account to test it in a real environment but based on the only "spec" I could find (https://www.ietf.org/archive/id/draft-murchison-sasl-login-00.txt), it should work fine (the main difference between the `PLAIN` auth is that we need to "feed" the username and password in separate responses).


If you want to give it a try, you can build PocketBase locally:

0. [Install Go 1.18+](https://go.dev/doc/install) (if you haven't already) 

1. Download/clone the PocketBase repo

2. Navigate to `examples/base`

3. Run `go build`

4. Start the generated executable as usual: `./base serve`

**Reply by SamuelQuinones** [reply]
it looks like that worked! Thank you for the quick response / fix time :) looking forward to the next full release
